client/db/interface.go

```diff
@@ -5,19 +5,31 @@ package db
 
 import (
 	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/encrypt"
 	"decred.org/dcrdex/dex/order"
 )
 
 // DB is an interface that must be satisfied by a DEX client persistent storage
 // manager.
 type DB interface {
 	dex.Runner
-	// Store allows the storage of arbitrary data.
-	Store(string, []byte) error
-	// Get retrieves values stored with Store.
-	Get(string) ([]byte, error)
-	// ValueExists checks if a value was previously stored.
-	ValueExists(k string) (bool, error)
+	// SetPrimaryCredentials sets the initial *PrimaryCredentials and core
+	// version for the client.
+	SetPrimaryCredentials(creds *PrimaryCredentials, coreVersion uint8) error
+	// UpdatePrimaryCredentials updates the *PrimaryCredentials.
+	UpdatePrimaryCredentials(creds *PrimaryCredentials) error
+	// PrimaryCredentials fetches the *PrimaryCredentials.
+	PrimaryCredentials() (*PrimaryCredentials, error)
+	// LegacyKeyParams get the app key parameters stored using the deprecated
+	// Store method.
+	LegacyKeyParams() ([]byte, error)
+	// UpgradeLegacyCredentials updates the stored credentials to the new
+	// *PrimaryCredential format and updates the stored core version.
+	UpgradeLegacyCredentials(creds *PrimaryCredentials, oldCrypter, newCrypter encrypt.Crypter,
+		coreVersion uint8) (walletUpdates map[uint32][]byte, acctUpdates map[string][]byte, err error)
+	// CoreVersion is the core version set with SetCoreVersion. If no core
```

Comment:
Good catch. As of right now, I'm allowing the core version to be passed in with the UpdateLegacyCredentials function since that allows the entire update to occur in a single transaction. We may need a `SetCoreVersion` method someday, but maybe I'll just update the docs for now.